src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -46,10 +46,30 @@ public void ConvertDayOfWeek()
             json = JsonSerializer.Serialize((DayOfWeek)(-1), options);
             Assert.Equal(@"-1", json);
 
-            // Not permitting integers should throw
             options = new JsonSerializerOptions();
             options.Converters.Add(new JsonStringEnumConverter(allowIntegerValues: false));
+            // Not permitting integers should throw
             Assert.Throws<JsonException>(() => JsonSerializer.Serialize((DayOfWeek)(-1), options));
+
+            // Numbers should throw
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<DayOfWeek>("1", options));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<DayOfWeek>("-1", options));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<DayOfWeek>(@"""1""", options));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<DayOfWeek>(@"""+1""", options));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<DayOfWeek>(@"""-1""", options));
+            
+            day = JsonSerializer.Deserialize<DayOfWeek>(@"""Monday""", options);
+            Assert.Equal(DayOfWeek.Monday, day);
+
+            // Numbers-formatted json string should also consider naming policy
```

Comment:
```suggestion
            // Numbers-formatted json string should first consider naming policy
```
Nice test case. Support for this is debatable but I think we should allow folks to work around validation issues here with a naming policy.

Will approve if no one opposes this @dotnet/area-system-text-json.

Reply:
Fixed.